lib/contextify.js

```diff
@@ -269,25 +318,46 @@ Decontextify.object = (object, traps, deepTraps, flags, mock) => {
 		// TypeError: 'getOwnPropertyDescriptor' on proxy: trap reported non-configurability for property '<prop>'
 		// which is either non-existant or configurable in the proxy target
 
+		let desc;
 		if (!def) {
 			return undefined;
 		} else if (def.get || def.set) {
-			return {
+			desc = {
+				__proto__: null,
 				get: Decontextify.value(def.get) || undefined,
 				set: Decontextify.value(def.set) || undefined,
 				enumerable: def.enumerable === true,
 				configurable: def.configurable === true
 			};
 		} else {
-			return {
+			desc = {
+				__proto__: null,
 				value: Decontextify.value(def.value),
 				writable: def.writable === true,
 				enumerable: def.enumerable === true,
 				configurable: def.configurable === true
 			};
 		}
+		if (!desc.configurable) {
+			try {
+				def = host.Object.getOwnPropertyDescriptor(target, prop);
+				if (!def || def.configurable) {
+					local.Reflect.defineProperty(target, prop, desc);
+				}
+			} catch (e) {
+				// Should not happen.
```

Comment:
What do you think about that

```js
(...)
} catch(e) {
   throwWhenInTest(e); // should not happen
}
(...)
```

The function `throwWhenInTest` could throw in your CI/local test environment and the errors would not go unnoticed.
Just a suggestion!

Reply:
First I think this can't be reached, since getOwnPropertyDescriptor target is never a proxy. def.configurable is allways a own data property and defineProperty can't fail since target can't be a proxy and the property is either non existant or configurable.
Second I don't know if the exception is from host or sandbox, most likely from the sandbox, so I don't want to expose it. It is possible to stringify it in another try catch and pass the string, but I don't know if that is realy necessary.